provider/kubernetes/kubernetes.go

```diff
@@ -135,6 +136,20 @@ func (p *Provider) Provide(configurationChan chan<- types.ConfigMessage, pool *s
 	return nil
 }
 
+func (p *Provider) shouldProcessIngress(ingressClass string) bool {
+	ingressClassRequired := "traefik"
```

Comment:
We could make a constant out of this like

```go
const ingressClassRequired = "traefik"
```

to indicate that this is not a string to be mutated.

That said, I don't think we need to variable anyway though if we decide to use the empty string as default value for `p.IngressClass`.

Reply:
@timoreimann you should not this but use a flag with a default (""). This allows the user to deploy multiple controllers with different class annotations to expose different ingress

Reply:
Ah sorry, late night logic flaw on my end. @aledbf is right, we need the default of `"traefik"`.

@yuvipanda please ignore my comments in this regard.

Reply:
Idea: Keep the default of `p.IngressClass` to `traefik` and still simplify the method implementation like this:

```go
func (p *Provider) shouldProcessIngress(ingressClass string) bool {
        return ingressClass == "" || ingressClass == p.IngressClass
}
```

I _think_ this _should_ work, though only tests will bring certainty...